src/main/java/org/candlepin/model/RhsmApiCompatCurator.java

```diff
@@ -103,4 +113,376 @@ public List<ConsumerEntitlementCount> getConsumerEntitlementCounts(Collection<St
+     * and ordered deterministically by ID and last check-in. For each consumer, relevant facts and
+     * installed products are extracted, and if the consumer represents a virtual machine (as indicated by
+     * the "virt.uuid" fact), an associated hypervisor is looked up by matching all possible UUID
+     * representations.
+     *
+     * @param owner         the {@link Owner} whose consumers are to be fetched; must not be null
+     * @param afterId       (optional) only consumers with an ID lexicographically greater than this value
+     *                      will be included
+     * @param afterUuid     (optional) only consumers with a UUID lexicographically greater than this value
+     *                      will be included
+     * @param afterCheckin  (optional) only consumers with a last check-in timestamp after this value will
+     *                      be included
+     * @param page          the page number to retrieve (1-based)
+     * @param perPage       the number of consumers to return per page
+     * @return a list of {@link ConsumerFeed} objects matching the specified filters, with facts, installed
+     *         products, add-ons, and hypervisor association (if applicable) fully populated
+     */
+    public List<ConsumerFeed> getConsumerFeeds(Owner owner, String afterId, String afterUuid,
+        OffsetDateTime afterCheckin, int page, int perPage) {
+
+        List<ConsumerFeed> consumersFiltered = getConsumersFiltered(owner.getOwnerKey(), afterId, afterUuid,
+            afterCheckin, page, perPage);
+        Map<String, HypervisorGuestMapping> hypervisorGuestMappings =
+            getHypervisorGuestMappingsByConsumer(owner.getOwnerId());
+        Map<String, Set<String>> addOnsByConsumer =
+            getAddOnsByConsumer(owner.getOwnerId(), afterId, afterUuid, afterCheckin, page, perPage);
+        Map<String, Map<String, String>> factsByConsumer =
+            getFactsByConsumer(owner.getOwnerId(), afterId, afterUuid, afterCheckin, page, perPage);
+        Map<String, Set<ConsumerFeedInstalledProduct>> installedProductsByConsumer =
+            getInstalledProductsByConsumer(owner.getOwnerId(), afterId, afterUuid, afterCheckin, page,
+                perPage);
+
+        for (ConsumerFeed consumerFeed : consumersFiltered) {
+            Map<String, String> facts = factsByConsumer.get(consumerFeed.getId());
+
+            // We must match the consumer's "virt.uuid" fact against all possible endianness variants,
+            // because guest IDs can be stored in partially reversed byte order in the database.
+            // By generating all possible UUID representations, we can correctly associate the consumer
+            // with its hypervisor if a mapping exists.
+            if (facts != null) {
+                String virtUuid = facts.get("virt.uuid");
+                if (virtUuid != null && !virtUuid.isEmpty()) {
+                    consumerFeed.setGuestId(virtUuid);
```

Comment:
I know you mentioned this to me today, but I wanted to add a comment here so we wouldn't forget. The guestId should only be set when there is a hypervisor for the guest system.